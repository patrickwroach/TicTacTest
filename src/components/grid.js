import React, { Component } from 'react';
import {Square} from './square';

export class Grid extends React.Component {

  renderSquare(i) {  
    return <Square status={this.props.grid.squares[i]} onClick={() => this.props.onSquareClick(i)} />;
  }


  render() {
  
    return (
    
         <div className="grid section">
          {this.renderSquare(0)}
          {this.renderSquare(1)}
          {this.renderSquare(2)}
          {this.renderSquare(3)}
          {this.renderSquare(4)}
          {this.renderSquare(5)}
          {this.renderSquare(6)}
          {this.renderSquare(7)}
          {this.renderSquare(8)}
        </div>
     
    );
  }
}
  