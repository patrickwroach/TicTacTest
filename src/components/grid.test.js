import React from 'react';
import { shallow } from 'enzyme';
import {Grid} from './grid';

describe('Grid', () => {
     
    const grid = {
        squares: Array(9).fill(null)
      };

  it('It should render correctly in "debug" mode', () => {  

    const component = shallow(<Grid 
        debug grid={grid}
         />);  
    expect(component).toMatchSnapshot();
  });
 
});


