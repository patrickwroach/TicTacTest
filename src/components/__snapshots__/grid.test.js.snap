// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid It should render correctly in "debug" mode 1`] = `
<div
  className="grid section"
>
  <Square
    onClick={[Function]}
    status={null}
  />
  <Square
    onClick={[Function]}
    status={null}
  />
  <Square
    onClick={[Function]}
    status={null}
  />
  <Square
    onClick={[Function]}
    status={null}
  />
  <Square
    onClick={[Function]}
    status={null}
  />
  <Square
    onClick={[Function]}
    status={null}
  />
  <Square
    onClick={[Function]}
    status={null}
  />
  <Square
    onClick={[Function]}
    status={null}
  />
  <Square
    onClick={[Function]}
    status={null}
  />
</div>
`;
